src/wasi.rs

```diff
@@ -1,9 +1,41 @@
-//! Implementation for WASI
+//! Implementation for WASI (preview 1 and 2)
+//!
+//! `target_env = "p1"` was introduced only in Rust 1.80, so on earlier compiler versions this
+//! code will result in a compilation error.
 use crate::Error;
 use core::mem::MaybeUninit;
-use wasi::random_get;
 
+#[cfg(not(any(target_env = "p1", target_env = "p2")))]
+compile_error!(
+    "Unknown version of WASI (only previews 1 and 2 are supported) \
+    or Rust version older than 1.80 was used"
+);
+
+#[cfg(target_env = "p1")]
 pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
-    unsafe { random_get(dest.as_mut_ptr().cast::<u8>(), dest.len()) }
+    unsafe { wasi::random_get(dest.as_mut_ptr().cast::<u8>(), dest.len()) }
         .map_err(|e| Error::from_os_error(e.raw().into()))
 }
+
+#[cfg(target_env = "p2")]
+pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
+    use wasi::random::random::get_random_u64;
+
+    let mut chunks = dest.chunks_exact_mut(size_of::<u64>());
+    for chunk in &mut chunks {
+        let dst: *mut u64 = chunk.as_mut_ptr().cast();
+        let val = get_random_u64();
```

Comment:
`get_random_u64` is used instead of `get_random_bytes` because the latter uses an allocation due to the Wit IDL [restrictions](https://github.com/WebAssembly/wasi-random/issues/27). This should be fine since the main use case of `getrandom` is seed generation.